### PR TITLE
chore: update OT core to 0.69.0

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
I'm not sure why `staticcheck` didn't find uses of deprecated `config.*Settings` fields.

There's one other problem here. Some our components use a [test package](https://github.com/open-telemetry/opentelemetry-collector/tree/v0.69.0/otelcol/otelcoltest) from upstream, and this test package has a dependency on the [`zpagesextension`](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.69.0/extension/zpagesextension/go.mod#L8), whose dependencies weren't upgraded in `0.69.0` core. I don't understand go packaging with multiple replaces that well, but the end result was that the tests wanted to use version `0.68.0` of that package, which isn't compatible with `0.69.0`. So I had to manually replace it in each component that uses it. Not sure if there's a more graceful way.

### CODE REVIEW

File: otelcolbuilder/cmd/configprovider.go [hunk 1]
```diff
@@ -38,8 +38,6 @@ import (
 func UseCustomConfigProvider(params *otelcol.CollectorSettings) error {
 	// feature flags, which are enabled by default in our distro
 	err := featuregate.GetRegistry().Apply(map[string]bool{
-		"receiver.apache.emitServerNameAsResourceAttribute":            true,
-		"receiver.apache.emitPortAsResourceAttribute":                  true,
 		"receiver.elasticsearch.emitClusterHealthDetailedShardMetrics": true,
 		"receiver.elasticsearch.emitAllIndexOperationMetrics":          true,
 	})
```
Comment (aboguszewski-sumo) [inline otelcolbuilder/cmd/configprovider.go:43]: The feature gates for elasticsearch are in beta now (enabled by default), so technically we could remove this, but I'm not sure if in this PR.

File: pkg/processor/k8sprocessor/go.mod [hunk 2]
```diff
@@ -104,3 +104,8 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
+
+// This component uses the otelcoltest package, which depends on zpagesextension, but uses a replace to refer to it in its go.mod
+// As a result, we pull the wrong version here without this replace
+// This can be removed once upstream updates the versions in zpagesextension go.mod or removes the dependency
+replace go.opentelemetry.io/collector/extension/zpagesextension => go.opentelemetry.io/collector/extension/zpagesextension v0.69.0
```
Comment (aboguszewski-sumo) [inline pkg/processor/k8sprocessor/go.mod:111]: Isn't this the same problem as with importing `sumologicextension` in `sumologicexporter`? I remember having similar issues when I used `sumologicexporter` externally. Idk if there is a reasonable solution for that, doesn't seem so.

**Review by aboguszewski-sumo — APPROVED** [review]
lgtm, but read my comment about feature gates

File: pkg/processor/k8sprocessor/go.mod (hunk 2 above)
Comment (swiatekm-sumo) [reply pkg/processor/k8sprocessor/go.mod:111]: There probably isn't, but we have a new release fixing this here: https://github.com/open-telemetry/opentelemetry-collector/releases/tag/cmd%2Fbuilder%2Fv0.69.1, so I'll update this PR.

Comment (swiatekm-sumo) [reply pkg/processor/k8sprocessor/go.mod:111]: Actually, it looks like 0.69.1 is not out for contrib, so we'd need to have two different versions for different components. I'll just leave this as is and add an issue to remove the replaces for 0.70.0.

Comment (swiatekm-sumo) [reply pkg/processor/k8sprocessor/go.mod:111]: https://github.com/SumoLogic/sumologic-otel-collector/issues/894

File: otelcolbuilder/cmd/configprovider.go (hunk 1 above)
Comment (aboguszewski-sumo) [reply otelcolbuilder/cmd/configprovider.go:43]: https://github.com/SumoLogic/sumologic-otel-collector/issues/896

**Pull request merged**